bridge/bridge.go

```diff
@@ -362,9 +362,27 @@ func (b *Bridge) remove(containerId string, deregister bool) {
 				log.Println("removed:", containerId[:12], service.ID)
 			}
 		}
+		cleanUpAttrs := func(services []*Service) {
+		    b.registry.AcquireDistributedLock()
+            distServices, err := b.registry.DistributedServices()
+            if err != nil {
+                log.Println("Failed to list distributed services: ", err)
+            }
+			for _, service := range services {
+				if _, ok := distServices[service.Name]; ok {
+                    log.Println("Service exists in another container... Attrs will be kept:", service.Name)
+                } else {
+                    b.registry.RemoveAttributes(service)
+                    log.Println("Service was not found elsewhere... Attrs will be removed:", service.Name)
+                }
+			}
+			b.registry.ReleaseDistributedLock()
```

Comment:
would it be better to use `defer` after line 366 to make sure the lock be released finally?
BTW, why need distributed lock here? I have a feeling that this lock does not help at all...